### ISSUE TITLE
OSC Patch Numbers

### ISSUE THREAD

**Issue by brodkin** [janslow/qlab-from-csv#29]
Hey @janslow, it's almost creepy how well you know exactly what software and features I need at any given moment. Thanks for creating these tools!

That said, I have a use case that hasn't quite been accounted for yet... We're trying your X32 development version, but our project has multiple OSC patches and there is no way to set the OSC patch to which the cues get assigned. By the time that we generate the cue list we have over 6,000 OSC cues and there's no way that we're going to manually change the OSC patch for each one.

Long story short it'd be great if there were a dropdown to select the OSC patch number for both X32 and ION cues.

**Comment by janslow** [comment]
Glad it's been useful for you!

That shouldn't be too hard to add, so I'll give it a go (assuming Apple's Xcode updates haven't broken everything again).

**Comment by brodkin** [comment]
You rock. Thx! 
I generally like to contribute code in these situations, but I don't know Swift. :(

**Comment by janslow** [comment]
That's fixed now in [1.3-x32](https://github.com/janslow/qlab-from-csv/releases/tag/1.3-x32). There's a new X32 sub-tab under Cues, which allows you to change what patch the X32 is on.

This project was a bit of an experiment into Swift, as it's not one of my normal languages. It was quite interesting to use, but Apple do keep on making breaking changes, and I don't think you can choose when to update the language (it's when Xcode updates), so it took longer for me to fix the new compile errors and get the old version working again than it took to add the OSC patch support.

**Comment by brodkin** [comment]
Thanks @janslow, you rock! We used this through the weekend and again yesterday and everything has been working great. If you're ever in California drop me a line--we'll treat you to a show!
